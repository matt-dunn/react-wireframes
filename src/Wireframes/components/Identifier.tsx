/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React from "react";
import styled from "@emotion/styled";

import { WireframeAnnotation } from "../api";

type IdentifierProps = {
  annotation: WireframeAnnotation;
  className?: string;
}

export const IdentifierContainer = styled.cite`
  all: initial;
  box-sizing: border-box;
  border-radius: 2em;
  color: #333;
  background-color: yellow;
  border-color: #caca00;
  white-space: nowrap;
  min-width: 2em;
  max-width: initial;
  min-height: auto;
  max-height: initial;
  height: 2em;
  padding: 0.25em 0.5em;
  margin: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  position: initial;
  overflow: visible;
  font-style: normal;
  font-weight: normal;
  font-size: 1rem;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  text-align: center;
  text-decoration: none;
  text-transform: none;
  flex-grow: 0;
  flex-shrink: 0;
`;

export const Identifier = ({ annotation, className }: IdentifierProps) => (
  <IdentifierContainer data-annotation-identifier className={className}>{annotation.id.toLocaleString()}</IdentifierContainer>
);
