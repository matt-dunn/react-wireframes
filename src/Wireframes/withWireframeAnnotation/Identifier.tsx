/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React from "react";
import styled from "@emotion/styled";
import { CSSTransition } from "react-transition-group";

import Identifier from "../components/Identifier";
import { WireframeAnnotation } from "../api";

type IdentifierProps = {
  annotation: WireframeAnnotation;
  show: boolean;
}

const IdentifierContainer = styled(Identifier)`
  position: absolute;
  top: -1em;
  left: -1em;
  z-index: 4000;
  cursor: default;
  transition: opacity 250ms, visibility 250ms;
  opacity: 0.75;
  visibility: visible;

  &.fade-Identifier.enter {
    opacity: 0;
    visibility: hidden;
  }
  &.fade-Identifier.enter-active {
    opacity: 0.75;
    visibility: visible;
  }
  &.fade-Identifier.exit {
    opacity: 0.75;
    visibility: visible;
  }
  &.fade-Identifier.exit-active {
    opacity: 0;
    visibility: hidden;
  }
`;

export const IdentifierComponent = ({ annotation, show = true }: IdentifierProps) => (
  <CSSTransition
    timeout={250}
    className="fade-Identifier"
    in={show}
    mountOnEnter
    unmountOnExit
  >
    <IdentifierContainer data-annotation-identifier annotation={annotation} />
  </CSSTransition>
);

export default IdentifierComponent;
