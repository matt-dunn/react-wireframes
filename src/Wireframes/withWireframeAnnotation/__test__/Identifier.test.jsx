/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React from "react";
import { shallow } from "enzyme";

import { IdentifierComponent as Component } from "../Identifier";

describe("Wireframe: Identifier", () => {
  it("should render correctly", () => {
    const wrapper = shallow(
      <Component
        annotation={{
          id: "123",
        }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
