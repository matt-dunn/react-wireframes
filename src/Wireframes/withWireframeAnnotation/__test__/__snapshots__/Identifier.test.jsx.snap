// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wireframe: Identifier should render correctly 1`] = `
<CSSTransition
  className="fade-Identifier"
  classNames=""
  in={true}
  mountOnEnter={true}
  timeout={250}
  unmountOnExit={true}
>
  <IdentifierContainer
    annotation={
      Object {
        "id": "123",
      }
    }
    data-annotation-identifier={true}
  />
</CSSTransition>
`;
